homeassistant/config_entries.py

```diff
@@ -316,7 +324,11 @@ async def async_remove(self, hass: HomeAssistant) -> None:
         if self.source == SOURCE_IGNORE:
             return
 
-        integration = await loader.async_get_integration(hass, self.domain)
+        try:
+            integration = await loader.async_get_integration(hass, self.domain)
+        except loader.IntegrationNotFound:
+            return
```

Comment:
Can we add a comment here too. 
```suggestion
            # The integration was likely a custom_component
            # that was uninstalled, or an integration
            # that has been renamed without removing the config
            # entry.
            return
```